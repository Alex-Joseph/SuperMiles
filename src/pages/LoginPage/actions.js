import { actionTypes } from './constants';


export const authenticateUser = (username, password, history) => (dispatch) => {
  if (username === 'username' && password === 'password') {
    dispatch(loginSuccess())
    history.push('/');
  } else {
    dispatch(loginFailure())
  }
};

export const logoutUser = history => (dispatch) => {
  history.push('/login');
  dispatch(logout())
};

export const loginSuccess = () => ({
  type: actionTypes.LOGIN_SUCCESS,
});

export const loginFailure = () => ({
  type: actionTypes.LOGIN_FAILURE,
});

export const logout = () => ({
  type: actionTypes.LOGOUT,
});
