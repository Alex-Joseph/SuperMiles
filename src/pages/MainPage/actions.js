import { actionTypes } from './constants';


export const postTransaction = (balance, product, history) => (dispatch) => {
  if (balance >= product.cost) {
    dispatch(updateBalance(balance - product.cost, product.desc));
    history.push('/receipt');
  } else {
    dispatch(insufficientBalance());
  }
};

export const updateBalance = (newBalance, desc) => ({
  type: actionTypes.UPDATE_BALANCE,
  balance: newBalance,
  transaction: { desc },
});

export const insufficientBalance = () => ({
  type: actionTypes.INSUFFICIENT_BALANCE,
});
