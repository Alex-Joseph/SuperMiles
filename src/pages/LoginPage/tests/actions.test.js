import { authenticateUser, loginSuccess, loginFailure, logout } from '../actions';


describe('action creators', () => {
  it('creates a authenticateUser action', () => {
    expect(authenticateUser()).toMatchSnapshot();
  });
  it('creates a loginSuccess action', () => {
    expect(loginSuccess()).toMatchSnapshot();
  });

  it('creates a loginFailure action', () => {
    expect(loginFailure()).toMatchSnapshot();
  });

  it('creates a logout action', () => {
    expect(logout()).toMatchSnapshot();
  });
});
