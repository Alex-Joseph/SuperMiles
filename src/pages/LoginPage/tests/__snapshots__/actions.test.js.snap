// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates a authenticateUser action 1`] = `[Function]`;

exports[`action creators creates a loginFailure action 1`] = `
Object {
  "type": "LOGIN_FAILURE",
}
`;

exports[`action creators creates a loginSuccess action 1`] = `
Object {
  "type": "LOGIN_SUCCESS",
}
`;

exports[`action creators creates a logout action 1`] = `
Object {
  "type": "LOGOUT",
}
`;
